### ISSUE TITLE
.zprint.edn not detected

### ISSUE THREAD

**Issue by MareinK** [kkinnear/zprint#210]
I am using zprint through the [vscode-clj-zprint extension](https://marketplace.visualstudio.com/items?itemName=rflagreca.vscode-clj-zprint) in VS Code/OSS Code. In the [zprint documentation](https://cljdoc.org/d/zprint/zprint/0.5.1/doc/documentation#how-to-configure-zprint), I found that:

> it will look in the current directory for a file `.zprintrc` and if it doesn't find one, it will look for `.zprint.edn`.

When I use an `rc` file in my project directory, my settings are used.
However when I try to use an `edn` file, my settings are not used for formatting.

The contents of my `edn`/`rc` file are:

```edn
{:map {:sort? false}}
```

So it seems like zprint does not actually look for `.zprint.edn`. Or am I doing something wrong? Thanks!

**Comment by kkinnear** [comment]
Thanks for asking!

The vs-clj-zprint extension (the existence of which was news to me) doesn't rely on zprint to find the various configuration files, probably because it is running in a Javascript (and therefore Clojurescrip) environment.  You can see from the code below:
```
(defn slurp
  [file]
  (->
    (.readFileSync fs file "utf8")
    (.toString)))

(defn get-zprintrc-file-str
  []
  (let [home-str (.homedir os)
        workspace-zprintrc-file-str (str (.. (first vscode/workspace.workspaceFolders) -uri -fsPath) "/.zprintrc")
        global-zprintrc-file-str (str home-str "/.zprintrc")]
    (cond
      (.existsSync fs workspace-zprintrc-file-str) workspace-zprintrc-file-str
      (.existsSync fs global-zprintrc-file-str) global-zprintrc-file-str
      :else nil)))

(defn get-config!
  []
  (let [zprintrc-file-str (get-zprintrc-file-str)]
    (try
      (when zprintrc-file-str
        (let [zprintrc-str (slurp zprintrc-file-str)]
          (when zprintrc-str (reader/read-string zprintrc-str))))
      (catch :default _ {}))))

(deftype ClojureDocumentRangeFormattingEditProvider [config]
 Object
   (provideDocumentRangeFormattingEdits [_ ^js document range options token]
     (let [text       (.getText document range)
           path vscode/window.activeTextEditor.document.uri.fsPath
           formatted (try (zprint/zprint-file-str text path config)
                                (catch js/Error e (js/console.log (.-message e))))]
       (when formatted #js [(vscode/TextEdit.replace range formatted)]))))
```
that it looks explicitly for `.zprintrc` files in the user's home directory or in the current directory.  This is rather different from the `:search-config? true` or the `:cwd-zprintrc? true` behavior of zprint itself in several ways.  Perhaps the most important is that it will use *either* the `.zprintrc` file in the current working directory *or* if none is found, the `.zprintrc` file in the user's home directory.  zprint will (if configured to do so), use both: applying the home directory `.zprintrc` file first, and then the `.zprintrc` file in the current working directory.  

I know little about the VS Code environment, but from what I can see at first glance, it appears that it is a Javascript (and therefore Clojurescript) environment, which means that the normal file access provided by Clojure isn't available.  Thus, to get any configuration at all into the extension, the code above had to get the information from the files "by hand", so to speak.

Basically, anything in the zprint documentation that discusses configuration file access should be ignored when using the vscode-clj-zprint extension.  I'm glad that the author provided some form of support for configuration files.  While it would be nicer if the configuration file handling mapped directly to that implemented for the Clojure version of zprint, that would be pretty involved, as the file handling is rather complex.

While it wouldn't be all that difficult for the extension to be modified to support `.zprint.edn` files as well as the `.zprintrc` files it now supports, until and unless that happens, you will have to use `.zprintrc` files exclusively.  In addition, placing all of your configuration in one of the two possible files is also required.

I hope that this helps!

**Comment by MareinK** [comment]
Hi, thank you for investigating and providing a very detailed answer! All is clear to me. I've created [an issue over at the vscode-clj-zprint repo](https://github.com/rflagreca/vscode-clj-zprint/issues/1) to see how the owner feels about it. I will continue to use `.zprintrc` for now.
